include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -32,30 +32,34 @@ namespace seqan3
 template <detail::config_element ... configs_t>
 class configuration;
 
-template <typename lhs_derived_t, typename lhs_value_t, typename rhs_derived_t, typename rhs_value_t>
-constexpr auto operator|(pipeable_config_element<lhs_derived_t, lhs_value_t> && lhs,
-                         pipeable_config_element<rhs_derived_t, rhs_value_t> && rhs)
+template <typename lhs_derived_t, typename lhs_value_t, bool lhs_needs_value,
+          typename rhs_derived_t, typename rhs_value_t, bool rhs_needs_value>
+constexpr auto operator|(pipeable_config_element<lhs_derived_t, lhs_value_t, lhs_needs_value> && lhs,
+                         pipeable_config_element<rhs_derived_t, rhs_value_t, rhs_needs_value> && rhs)
 {
     return configuration{static_cast<lhs_derived_t &&>(lhs)}.push_back(static_cast<rhs_derived_t &&>(rhs));
```

Comment:
This will be `undefined behaviour` if the given config `lhs` is not of type `lhs_derived_t`.

Reply:
Is there a solution for this that I should include in this PR?
Maybe we can assert or require the type dependency?